lib/new_relic/application.ex

```diff
@@ -11,7 +11,7 @@ defmodule NewRelic.Application do
     children = [
       worker(NewRelic.Logger, []),
       supervisor(NewRelic.AlwaysOnSupervisor, []),
-      supervisor(NewRelic.EnabledSupervisor, []),
+      worker(NewRelic.EnabledSupervisorManager, []),
```

Comment:
This might be too much, but should this be a DynamicSupervisor, and the EnabledSupervisor would then be a child of it when started? Or do we feel like `start_link()` is enough?

Reply:
Without more than one child, it probably doesn't make sense. Answered my own question. Ignore me 😝

Reply:
That's an interesting thought, I went this direction so that the manager process could send a message and get one back from `AgentRun` to signal when to start the `EnabledSupervisor`. The original problem was that there was a blocking call in `start_link` of `EnabledSupervisor`, so this made it async.

It would be nice if it was all supervisors w/o a `GenServer` stuck in between, perhaps `AgentRun` could trigger the `EnabledSupervisor` to start as part of it's connect process

Reply:
Tried it out in https://github.com/newrelic/elixir_agent/pull/187 , I like that it uses standard supervisors the whole way down